crates/matrix-sdk/src/widget/machine/from_widget.rs

```diff
@@ -174,22 +179,15 @@ pub(super) enum ApiVersion {
 }
 
 #[derive(Deserialize, Debug)]
-#[serde(untagged)]
-pub(super) enum ReadEventRequest {
-    ReadStateEvent {
-        #[serde(rename = "type")]
-        event_type: StateEventType,
-        state_key: StateKeySelector,
-    },
-    ReadMessageLikeEvent {
-        #[serde(rename = "type")]
-        event_type: MessageLikeEventType,
-        limit: Option<u32>,
-    },
+pub(super) struct ReadEventsRequest {
+    #[serde(rename = "type")]
+    pub(super) event_type: TimelineEventType,
+    pub(super) state_key: Option<StateKeySelector>,
+    pub(super) limit: Option<u32>,
```

Comment:
This will badly break the data format, and I suppose that's the reason why you're asking for a coordinated release; is there a chance we could support both the previous and new formats, instead?